Level-2/her/her.cpp

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright (c) 2021 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "helpers.hpp"
+#include <hip/hip_runtime.h>
+#include <hip/hip_complex.h>
+#include <math.h>
+#include <rocblas.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <vector>
+#include <complex>
+
+int main(int argc, char** argv)
+{
+    helpers::ArgParser options("Nax");
+    if(!options.validArgs(argc, argv))
+        return EXIT_FAILURE;
+
+    rocblas_status rstatus = rocblas_status_success;
+
+    rocblas_int N    = options.N;
+    rocblas_int incx = options.incx;
+
+    float hAlpha = options.alpha;
+
+    const rocblas_fill uplo = rocblas_fill_upper;
+
+    size_t sizeX, absIncx;
+
+ 
+    rocblas_int lda   = N;
+    size_t      sizeA = lda * size_t(N);
+
+    absIncx = incx >= 0 ? incx : -incx;
+
+    sizeX = N * absIncx;
+
+    // Naming: dK is in GPU (device) memory. hK is in CPU (host) memory
+
+    std::vector<hipFloatComplex> hA(sizeA);
+
+    // we are using std::complex for it's operators and it has same memory layout 
+    // as hipFloatComplex so can copy the data into the array for use in the rocblas C API
+    std::vector<std::complex<float>> hX(sizeX, std::complex<float>(2.0f, 0.5f));
+
+    std::vector<hipFloatComplex> hAGold(sizeA); 
+
+    // initialize simple data for simple host side reference computation
+    helpers::matIdentity(hA.data(), N, N, lda);
```

Comment:
Is there a reason matrix `A` is an identity matrix? Could it be initialized with random values in the upper triangular part?

Reply:
Well it is identity just so I didn't have to write a complex helper for Hermetian matrix fill.  I wanted to just focus on types and the conversion to what is required for the function call, nothing to do with really testing the functionality or explaining what a Hermetian matrix is.  I met you half way by adding a vector complex random initializer, but if really desired I can add a Hermetian helper and fix the CPU reference to handle it.   Maybe the reference code needs a disclaimer to say it isn't the full her implementation?